### ISSUE TITLE
TS build fails after upgrade to 7.3.0 with noImplicitAny flag.

### ISSUE THREAD

**Issue by raveclassic** [swimlane/ngx-datatable#671]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here
```

**Current behavior**

TS fails building `release/components/row-detail/row-detail.directive.d.ts` with `--noImplicitAny` flag with `(10, 34) Parameter 'index' implicitly has an 'any' type.` error.

**Expected behavior**

Build runs without errors.

**Reproduction of the problem**

- update to `@swimlane/ngx-datatable@7.3.0` in any project using ngx-datatable
- build the project

**What is the motivation / use case for changing the behavior?**

Fix the bug

**Please tell us about your environment:**

Win10, Node 7.7.3, Npm 4.1.2

* **Table version:** 0.7.x

0.7.3

* **Angular version:** 2.0.x

4.0.1

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

Not related to browsers

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
TS 2.2.2

**Comment by amcdnl** [comment]
You need to have typesscript ignore node_modules.

**Comment by raveclassic** [comment]
How is related to `node_modules`? It's an expected behavior for TS to build generated `.d.ts` schemes from `ngx-datatable` because of `import { NgxDatatableModule } from '@swimlane/ngx-datatable';` in consuming project.

Default ng2 app configuration does not require excluding `node_modules` because app is build with `src/tsconfig.app.json` inside `src`, that's why `node_modules` are not included anyway. 

The bug is a regression after [this change](https://github.com/swimlane/ngx-datatable/commit/0c703af55c354bd3c2e84cf8674671cdd93c4d3b#diff-66faf60ab49dbc61f1e168c82859c37dR14)

**Comment by jinder** [comment]
Also seeing this error.  Have 'fixed' it by setting skipLibCheck to true in tsconfig.  It would be nice to consume the library in the upcoming 'strict' mode (with noImplicitAny and strictNullChecks).

**Comment by QuentinFchx** [comment]
I also have the issue, preventing me from upgrading the library

**Comment by SimonSchick** [comment]
Agreeing with @raveclassic I don't see why removing `any` did do any good here. Ideally you want to enable `noImplicitAny` in your project as well.

**Comment by amcdnl** [comment]
@raveclassic - I removed that implicit but there is implicit `any` used quite a bit throughout the project and has been for a very long time so I'm not sure why this is just showing up now. Perhaps you upgraded TS and its enforcing things more strictly?

I've always excluded node_modules from compilation for perf reasons and headache reasons.

I did go ahead and fix that specific instance but there is violations everywhere of that rule, so I'm not sure if it will help. Let me know if that fixes your issue.
